cas-server-webapp-actions-aup-ldap/src/main/java/org/jasig/cas/web/flow/LdapAcceptableUsagePolicyRepository.java

```diff
@@ -0,0 +1,177 @@
+                        Boolean.TRUE.toString()));
+                for (final LdapAttribute attr : entry.getAttributes()) {
+                    mods.add(new AttributeModification(AttributeModificationType.REPLACE, attr));
+                }
+                final ModifyRequest request = new ModifyRequest(currentDn,
+                        mods.toArray(new AttributeModification[]{}));
+                operation.execute(request);
+                return true;
+            } catch (final LdapException e) {
+                logger.error(e.getMessage(), e);
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Search for service by id.
+     *
+     * @param connection the connection
+     * @param id         the id
+     * @return the response
+     * @throws LdapException the ldap exception
+     */
+    private Response<SearchResult> searchForId(final Connection connection, final String id)
+            throws LdapException {
+
+        final SearchFilter filter = new SearchFilter(this.searchFilter);
+        filter.setParameter(0, id);
+        return executeSearchOperation(connection, filter);
+    }
+
+    /**
+     * Gets connection from the factory.
+     * Opens the connection if needed.
+     *
+     * @return the connection
+     * @throws LdapException the ldap exception
+     */
+    private Connection getConnection() throws LdapException {
+        final Connection c = this.connectionFactory.getConnection();
+        if (!c.isOpen()) {
+            c.open();
+        }
+        return c;
+    }
+
+    /**
+     * Checks to see if response has a result.
+     *
+     * @param response the response
+     * @return true, if successful
+     */
+    private boolean hasResults(final Response<SearchResult> response) {
+        final SearchResult result = response.getResult();
+        if (result != null && result.getEntry() != null) {
+            return true;
+        }
+
+        logger.trace("Requested ldap operation did not return a result or an ldap entry. Code: {}, Message: {}",
+                response.getResultCode(), response.getMessage());
+        return false;
+    }
+
+
+    /**
+     * Execute search operation.
+     *
+     * @param connection the connection
+     * @return the response
+     * @throws LdapException the ldap exception
+     */
+    private Response<SearchResult> executeSearchOperation(final Connection connection, final SearchFilter id)
+            throws LdapException {
+
+        final SearchOperation searchOperation = new SearchOperation(connection);
+        final SearchRequest request = newSearchRequest(id);
+        logger.debug("Using search request {}", request.toString());
+        return searchOperation.execute(request);
+    }
+
+    /**
+     * Builds a new request.
+     *
+     * @return the search request
+     */
+    private SearchRequest newSearchRequest(final SearchFilter filter) {
```

Comment:
And I also wonder about those "generic" LDAP methods in there like `getConnection`, `searchForId`, `executeSearchOperation`, `hasResult` - may be good candidates to lift those into an Ldap utils or a reusable component, leaving only the business logic code in here utilizing those infrastructural components. And who knows - those could be reusable. I could think of one use case with CAS customization that I was working where I'd need to use Ldaptive search capabilities and this would be very useful.